### PR TITLE
feat: restrict sending gnot by default

### PR DESCRIPTION
**Pull request by deelawn**
_NOTE_: I've just seen the comment from @moul on the original issue related to this, so parts of this solution may need to be reworked -- https://github.com/gnolang/gno/issues/1837#issuecomment-2018493517

BREAKING CHANGE in the sense that GNOT will no longer be transferable by default unless otherwise specified when starting the node.

This PR aims to do the following:
- Disable explicit GNOT transfers by default (not default for gnodev and txtar tests)
- GNOT _can only_ be transferred to pay gas fees

This is the first of two PRs. The next will be to add a mechanism to unlock GNOT transfer restrictions on a per account basis by the account that wishes to be unlocked broadcasting a new type of transaction.

I'm seeking feedback on this before merging, mainly to get some clarification on the following questions:
- Should locking be enabled by default?  I think it makes sense, as genesis can be modified or a flag can be specified when starting a node when doing lazy genesis initialization.
- Should there also be fields in genesis that are exempt from the restrictions, such as some system account owned by us?  The subsequent unlocking PR will also likely include "semi-exempt" accounts as genesis fields -- these being accounts that are exempt in the sense that they are _able_ to submit a transaction to unlock their account.  Genesis will also have a field specifying whether or not accounts are generally unlockable or not.  A chain upgrade would be required to make accounts generally unlockable after initial launch

No docs will be updated as a part of this PR until there has been more of a discussion.

<details><summary>Contributors' checklist...</summary>

- [x] Added new tests, or not needed, or not feasible
- [x] Provided an example (e.g. screenshot) to aid review or the PR is self-explanatory
- [ ] Updated the official documentation or not needed
- [x] No breaking changes were made, or a `BREAKING CHANGE: xxx` message was included in the description
- [ ] Added references to related issues and PRs
- [x] Provided any useful hints for running manual tests
- [x] Added new benchmarks to [generated graphs](https://gnoland.github.io/benchmarks), if any. More info [here](https://github.com/gnolang/gno/blob/master/.benchmarks/README.md).
</details>

### CODE REVIEW

File: gno.land/cmd/gnoland/genesis_generate.go [hunk 1]
```diff
@@ -20,6 +20,7 @@ type generateCfg struct {
 	blockMaxDataBytes int64
 	blockMaxGas       int64
 	blockTimeIota     int64
+	gnotUnrestricted  bool
```
Comment (leohhhn) [inline gno.land/cmd/gnoland/genesis_generate.go:23]: Could it be better to replace `gnot` with `ugnot` in code for accuracy?

Comment (deelawn) [reply gno.land/cmd/gnoland/genesis_generate.go:23]: I can. Keep in mind this is still marked as draft and I plan on changing some things.

File: gno.land/cmd/gnoland/dylan/send.txtar [hunk 2]
```diff
@@ -0,0 +1,4 @@
+gnoland start --ugnot-locked
+
+! gnokey maketx send -send "9999999ugnot" -to g1jg8mtutu9khhfwc4nxmuhcpftf0pajdhfvsqf5 -gas-fee 1ugnot -gas-wanted 10000000 -broadcast -chainid=tendermint_test test1
+stderr 'account is locked'
```
Comment (moul) [inline gno.land/cmd/gnoland/dylan/send.txtar:4]: "ugnot transfer is locked" (the account it not locked)

**Comment by moul** [comment]
Sorry, I just left one comment, but I think before reviewing individual lines, we need:

1. To clarify the overall strategy.
2. To ensure the names will be good enough, especially if we want to make changes in TM2 and not only gno.land.

---

Braindump:

- We need a way to lock GNO token transfer globally, which can be changed by a DAO, and a way to enforce whitelist exceptions (e.g., faucets) during that period.
- We could also need or want another layer of whitelisting, where this time, it won't be an exception during global lock, but something that will only works when token transfers are globally unlocked. This would allow differentiating accounts that have "signed the T.O.S/EULA/whatever" from non-signers.
- In terms of implementation, we have already identified other needs that may require a way to set a global flag for the chain from a contract, which can be read (for cheap) by TM2. This introduces the notion of adding a `std.SetConfig(key, value string)` function, where the chain and eventually other contracts could call `std.GetConfig(realm, key string) string`. Alternatively, we could rely on modules where gno.land will load the GNO VM to call a contract helper, as done for instance in #2229.
- this global mechanism could potentially be used not only for global variables, but also for account-specific or realm-specific variables, by just prefixing the key with what we're looking for, i.e., `gno.land/r/sys/tos.SetConfig("signed-by-<std.GetOrigCaller()>")`
- We're considering extending the account store. Here, we have two main location options: one is the TM2 store itself, and another could be specific to gno.land. If we want to use the TM2 store, there's a question about minimalism and staying generalist that we may eventually address by adding optional columns that can be used by gnovm-based chains to add new fields on the account (i.e., USER1, USER2, USER3 in linux processes), and then add some helpers or pass an account keeper to apply the changes. If we prefer using a dedicated gno.land store for users, then we have a constraint of performance, where ideally, we should do a single store read query and not start having two.

**Pull request closed without merging**